src/main.ts

```diff
@@ -0,0 +1,178 @@
+      // eslint-disable-next-line no-sync
+      child_process.execSync('yarn --silent --json workspaces info', {
+        cwd: directory,
+        encoding: 'utf-8',
+      })
+    ).data
+  )
+
+  const result: string[] = []
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+  for (const key of Object.keys(json)) {
+    const location = 'location'
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    if (json[key][location] !== undefined) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access
+      result.push(json[key][location])
+    }
+  }
+  return result
+}
+
+export function index(options: Options): void {
+  let config = ts.parseCommandLine(
+    ['-p', options.project],
+    (relativePath: string) => path.resolve(options.project, relativePath)
+  )
+  let tsconfigFileName: string | undefined
+  if (config.options.project) {
+    const projectPath = path.resolve(config.options.project)
+    if (ts.sys.directoryExists(projectPath)) {
+      tsconfigFileName = path.join(projectPath, 'tsconfig.json')
+    } else {
+      tsconfigFileName = projectPath
+    }
+    if (!ts.sys.fileExists(tsconfigFileName)) {
+      console.error(
+        `skipping project '${options.project}' because it's missing tsconfig.json (expected at ${tsconfigFileName})`
+      )
+      return
+    }
+    config = loadConfigFile(tsconfigFileName)
+  }
+
+  if (config.fileNames.length === 0) {
+    console.error('no input files')
+    process.exitCode = 1
+  }
+
+  new Indexer(config, options).index()
+}
+
+if (require.main === module) {
+  main()
+}
+
+function loadConfigFile(file: string): ts.ParsedCommandLine {
+  const absolute = path.resolve(file)
+
+  const readResult = ts.readConfigFile(absolute, path => ts.sys.readFile(path))
+  if (readResult.error) {
+    throw new Error(
+      ts.formatDiagnostics([readResult.error], ts.createCompilerHost({}))
+    )
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const config = readResult.config
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  if (config.compilerOptions !== undefined) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+    config.compilerOptions = {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      ...config.compilerOptions,
+      ...defaultCompilerOptions(file),
+    }
+  }
+  const result = ts.parseJsonConfigFileContent(
+    config,
+    ts.sys,
+    path.dirname(absolute)
+  )
+  if (result.errors.length > 0) {
+    throw new Error(
+      ts.formatDiagnostics(result.errors, ts.createCompilerHost({}))
+    )
+  }
+  return result
+}
+
+function defaultCompilerOptions(configFileName?: string): ts.CompilerOptions {
+  const options: ts.CompilerOptions =
+    configFileName && path.basename(configFileName) === 'jsconfig.json'
```

Comment:
Typo?
```suggestion
    configFileName && path.basename(configFileName) === 'tsconfig.json'
```

Reply:
Added a comment, jsconfig.json is an actual thing https://sourcegraph.com/search?q=context:global+file:jsconfig.json&patternType=literal

This code was adapted from the original lsif-node implementation.